### DISCUSSION TITLE
Import/export the entire DB

### DISCUSSION THREAD

**Discussion by glensimister** [directus/directus#14763] [Archive]
I've only just stumbled across Directus. It seems like a great product. Well done, and thanks for making it open source. I have a few questions...

I noticed that you can import/export articles, users and files on the cloud version. However, i want to import/export the entire database. Is there a way to do this? 

Another thing, I'm using the demo data and using the API i can retrieve the article id, status, etc. However, I'm not able to see the title, summary or body fields in the response. Any idea why i can't see these fields?

Can Directus generate RSS feeds?  

Are you planning to offer hosting for front-ends, with custom domains etc? 

Cheers

**Comment by rijkvanzanten** [comment]
> I noticed that you can import/export articles, users and files on the cloud version. However, i want to import/export the entire database. Is there a way to do this?

There's currently built in way to dump/restore the whole database, though there's various ways to achieve that without having to rely on Directus in the first place (most database vendors even have a dump/restore command built in). 

> Another thing, I'm using the demo data and using the API i can retrieve the article id, status, etc. However, I'm not able to see the title, summary or body fields in the response. Any idea why i can't see these fields?

Most likely that your current user (or the public role if you're not using an access token) doesn't have permission to read those fields.

> Can Directus generate RSS feeds?

Directus is "headless", meaning that there's no project output built-in (by design). That being said, it's fairly straightforward to generate an RSS feed with something like [eleventy](https://11ty.dev). 

> Are you planning to offer hosting for front-ends, with custom domains etc?

While it's definitely interesting, it's not the primary focus of Directus as a platform at this point 👍🏻

**Comment by glensimister** [comment]
OK thanks for the reply. 

I'm using Directus cloud and I assume you are referring to a self-hosted instance, right? That's fine. If I need to self-host Directus then i'll do that instead. Is there any documentation about how to "dump/restore the whole database"?  

"Most likely that your current user (or the public role if you're not using an access token) doesn't have permission to read those fields".

Thanks, but i don't think that's the problem. I have already set the articles permissions to public (green tick) and I have also tried with an access token. Any other suggestions?

**Reply by rijkvanzanten** [reply]
> I'm using Directus cloud and I assume you are referring to a self-hosted instance, right? That's fine. If I need to self-host Directus then i'll do that instead. Is there any documentation about how to "dump/restore the whole database"?

Gotcha, that wasn't clear to me from your original message 🙂 From a cloud service perspective, we're currently working support for accessing the underlying raw database (at which point you can effectively do whatever you want with it). In the meantime, the cloud team can provide dumps by requesting it through the chat widget on https://directus.cloud

> Thanks, but i don't think that's the problem. I have already set the articles permissions to public (green tick) and I have also tried with an access token. Any other suggestions?

Hard to tell without more information about the configuration and access rules of the project 👍🏻

**Comment by glensimister** [comment]
OK thanks again. BTW I just figured out that the reason why the fields weren't showing is because they were saved as translations (?). Anyway, after creating my own fields (i'm using the demo data), i can see them in the response ;) 

The reason I'm asking about importing/exporting the DB is because i'm trying to scope out whether Directus is suitable for my project. In the same way that you can install dummy data when you sign up, i would like my users to create their own Directus instance, and be able to install a data-set that i have created. However, i can't assume that they will have any technical knowledge. I suspect it is done via the command-line, in which case I will have to find another way.

**Reply by rijkvanzanten** [reply]
We're currently actively working on https://github.com/directus/directus/issues/11481 which implements a set of new endpoints to easily export/import/migrate schema changes for that type of use case 🙂

**Reply by glensimister** [reply]
OK that sounds good ;)
